tests/dags/providers/provider_api_scripts/test_cleveland_museum.py

```diff
@@ -35,197 +35,177 @@ def validate_url_string():
-    with patch.object(clm.delay_request, "get", return_value=r) as mock_get:
-        response_json, total_images = clm._get_response(query_param)
-    expected_response = _get_resource_json("response_no_data.json")
-
-    assert mock_get.call_count == 1
-    assert response_json == expected_response
-    assert total_images == 0
-
-
-def test_get_response_failure():
-    query_param = {"cc": 1, "has_image": 1, "limit": 1, "skip": -1}
-    r = requests.Response()
-    r.status_code = 500
-    r.json = None
-    with patch.object(clm.delay_request, "get", return_value=r) as mock_get:
-        clm._get_response(query_param)
-
-    assert mock_get.call_count == 3
-
-
-def test_handle_single_response():
-    response_json = _get_resource_json("response_success.json")
-    batch = response_json["data"]
-    # Patching save_item because it will have the filetype after clean_metadata
-    with patch.object(clm.image_store, "save_item") as mock_save_item:
-        clm._handle_response(batch)
-    expected_image = {
-        "creator": "",
-        "foreign_identifier": "96887",
-        "foreign_landing_url": "https://clevelandart.org/art/1916.586.a",
-        "width": 641,
-        "height": 900,
-        "filesize": 222248,
-        "filetype": "jpg",
-        "url": "https://openaccess-cdn.clevelandart.org/1916.586.a/1916.586.a_web.jpg",
-        "license_": CC0_LICENSE.license,
-        "license_version": CC0_LICENSE.version,
-        "meta_data": {
-            "license_url": CC0_LICENSE.url,
-            "raw_license_url": None,
-            "accession_number": "1916.586.a",
-            "classification": "Miscellaneous",
-            "credit_line": "Gift of Mr. and Mrs. J. H. Wade",
-            "culture": "Germany, 18th century",
-            "date": "1700s",
-            "tombstone": "Scent Bottle, 1700s. Germany, 18th century. Glass with "
-            "enamel decoration; overall: 10.2 cm (4 in.). The Cleveland "
-            "Museum of Art, Gift of Mr. and Mrs. J. H. Wade 1916.586.a",
-            "technique": "glass with enamel decoration",
-        },
-        "title": "Scent Bottle",
-    }
-
-    actual_image = mock_save_item.call_args[0][0]
-    for key, value in expected_image.items():
-        assert getattr(actual_image, key) == value
-
-
-def test_get_response_None():
-    query_param = {"cc": 1, "has_image": 1, "limit": 1, "skip": -1}
-    with patch.object(clm.delay_request, "get", return_value=None) as mock_get:
-        response_json, total_images = clm._get_response(query_param)
-
-    assert response_json is None
-    assert total_images == 0
-    # Retries
-    assert mock_get.call_count == 3
-
-
-def test_handle_response():
-    response_json = _get_resource_json("handle_response_data.json")
-    data = response_json["data"]
-    actual_total_images = clm._handle_response(data)
-    expected_total_images = 100
-
-    assert actual_total_images == expected_total_images
+class TestClevelandDataIngester(unittest.TestCase):
```

Comment:
I am not entirely sure, but I think we're mostly using pytest in the catalog. And I keep hearing in podcasts how great it is compared to other testing frameworks. Is unittest better for this case?

Reply:
Not really, but we're not doing much with it either way -- just formatting, basically 😆